### DISCUSSION TITLE
onDisconnect hook trigger events

### DISCUSSION THREAD

**Discussion by kvindascr** [enisdenjo/graphql-ws#324] [Q&A] [answered]
I have a scenario where I need to identify if one of the clients connected to a subscription has disconnected from the WS

This made me wonder a couple of questions regarding the onDisconnect hook (server-side).

Our connection uses the `keepAlive` and `retryAttempts` params.

Based on that configuration I was wondering if the `onDisconnect` (server-side) was triggered on any disconnection from the WS, or if it would be triggered on pong not being received? or maybe both?

For example, what would occur in the case that a client briefly loses connections and gains connection again? is the onDisconnect triggered on the server-side? Would the same occurs if there is some lag in the connection?

Any information to clarify this scenario is welcome.

Thanks!

**Comment by enisdenjo** [comment, marked as answer]
Please read the documentation for [`onDisconnect`](https://github.com/enisdenjo/graphql-ws/blob/master/docs/interfaces/server.ServerOptions.md#ondisconnect) and then also for [`onClose`](https://github.com/enisdenjo/graphql-ws/blob/master/docs/interfaces/server.ServerOptions.md#onclose). All your questions are answered within.
